src/memory.jl

```diff
@@ -100,6 +99,56 @@ Returns the used amount of memory (in bytes), allocated on the agent.
 """
 used() = total()-free()
 
+"""
+    pointerinfo(ptr::Ptr)
+    pointerinfo(buf::Buffer)
+    pointerinfo(a::Array)
+
+Retrieve information about the allocation referenced by the given pointer.
+"""
+function pointerinfo(ptr::Ptr)
+    ptrinfo = Ref{HSA.PointerInfo}()
+
+    HSA.amd_pointer_info(Ptr{Cvoid}(ptr), ptrinfo, C_NULL, Ptr{UInt32}(C_NULL), C_NULL) |> check
+
+    return ptrinfo[]
+end
+pointerinfo(buf::Buffer) = pointerinfo(buf.ptr)
+pointerinfo(a::Array) = pointerinfo(pointer(a))
+
+## Page-locking
+
+"""
+    lock(ptr::Ptr, bytesize::Integer, agent::HSAAgent)
+    lock(ptr, bytesize)
+    lock(a::Array, agent)
+    lock(a)
+
+Page-lock a host pointer allocated by the OS allocator and return a new pointer from
+the given `agent`. For more information, see `hsa_amd_memory_lock()`.
+
+See also: [`unlock`](@ref)
+"""
+function lock(ptr::Ptr, bytesize::Integer, agent::HSAAgent)
+    plocked = Ref{Ptr{Cvoid}}()
+    HSA.amd_memory_lock(Ptr{Cvoid}(ptr), bytesize, Ref(agent.agent), 1, plocked) |> check
+    return plocked[]
+end
+lock(ptr, bytesize) = lock(ptr, bytesize, get_default_agent())
+lock(a::Array, agent::HSAAgent) = lock(pointer(a), sizeof(a), agent)
```

Comment:
Needs a test for this method

Reply:
Hmm, you mean testing if the memory is indeed accessible from the right agent?
Unfortunately, I've found no good way to test that since the agent owner returned by `pointerinfo` is the CPU regardless of the page-locking, at least on my system.

Reply:
No, just a test that calls this method. I didn't see anywhere that this specific method was called, but I could be mistaken.

If we can't test actual ownership, I'm ok with that. Some things just don't lend themselves well to testing :smile: